core/modules/parsers/wikiparser/rules/syslink.js

```diff
@@ -0,0 +1,45 @@
+/*\
+title: $:/core/modules/parsers/wikiparser/rules/syslink.js
+type: application/javascript
+module-type: wikirule
+
+Wiki text inline rule for system tiddler links.
+Can be suppressed preceding them with `~`.
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+exports.name = "syslink";
+exports.types = {inline: true};
+
+exports.init = function(parser) {
+	this.parser = parser;
+	// Regexp to match
+	this.matchRegExp = /~?(?:\$):[^\s<>{}\[\]`|'"\\^~]+(?:\/|\b)/mg;
```

Comment:
Didn't further inspect where something breaks. I just noticed that when looking at a system tiddler, the title would be rendered with a bunch of broken html with the first simplified regexp above, so I readded those angle brackets to the regexp which fixed that problem.